app/tut/ga/cluiche.md

```diff
@@ -0,0 +1,701 @@
+```{.setanta .numberLines}
+gníomh bog_liathróid() {
+    x_liathróide += dx_liathróide
+    y_liathróide += dy_liathróide
+}
+```
+
+Cruthaímis 2 athróg nua inár gníomh: `x_tuartha` agus `y_tuartha`. Bainfimid úsáid astu chun suíomh
+na liathróide atá [[tuartha|predicted]] a stóráil, ansin b'fhéidir linn iad a úsáid chun seic a
+dhéanamh an mbeidh an liathróid thar imill na stáitse. Seo é ár gcód nua, ní dhéanann sé aon rud
+difriúil ach tá spás ann chun seiceanna a dhéanamh níos déanaí.
+
+```{.setanta .numberLines}
+gníomh bog_liathróid() {
+    >-- Cá bhfuil an liathróid ag dul?
+    x_tuartha := x_liathróide + dx_liathróide
+    y_tuartha := y_liathróide + dy_liathróide
+
+    >-- Déanfaimid ár seiceanna anseo
+
+    >-- Tar éis na seiceanna, úasdátaigh áit na liathróide.
+    x_liathróide += dx_liathróide
+    y_liathróide += dy_liathróide
+}
+```
+
+Anois tá sé in am tosú ag scríobh ár loighic preibe. Chun an liathróid a chasadh timpeall sa treo x,
+níl le déanamh againn ach an luach `dx_liathróide` a athrú go [[diúltach|negative]]. Mar an gcéanna,
+chun an liathróid a chasadh timpeall sa treo y, athraímid `dy_liathróide` go `-dy_liathróide`.
+
+Má tá an liathróid ar tí dul thar bun an stáitse, tá an cluiche thart. Is féidir linn an gníomh
+`stop()` a úsáid chun an ríomhchlár a stopadh.
+
+Seo an cód chun é sin a dhéanamh:
+
+```{.setanta .numberLines}
+gníomh bog_liathróid() {
+    >-- Cá bhfuil an liathróid ag dul?
+    x_tuartha := x_liathróide + dx_liathróide
+    y_tuartha := y_liathróide + dy_liathróide
+
+    >-- Déanfaimid ár seiceanna anseo
+
+    má x_tuartha < 0 {
+        >-- Beidh an liathróid thar an taobh chlé.
+        >-- Cas an liathróid timpeall sa treo x.
+        dx_liathróide = -dx_liathróide
+    }
+    má x_tuartha > fad_x@stáitse {
+        >-- Beidh an liathróid thar an taobh dheis.
+        >-- Cas an liathróid timpeall sa treo x.
+        dx_liathróide = -dx_liathróide
+    }
+    má y_tuartha < 0 {
+        >-- Beidh an liathróid thar barr an stáitse.
+        >-- Cas an liathróid timpeall sa treo y.
+        dy_liathróide = -dy_liathróide
+    }
+    má y_tuartha > fad_y@stáitse {
+        >-- Tá an liathróid ag dul thar bun an stáitse.
+        >-- Tá an cluiche thart.
+        scríobh("CLUICHE THART")
+        stop()
+    }
+
+    >-- Tar éis na seiceanna, úasdátaigh áit na liathróide.
+    x_liathróide += dx_liathróide
+    y_liathróide += dy_liathróide
+}
+```
+
+## Preab as an slacán
+
+Anois preabann an liathróid as na ballaí, ach fós téann sé díreach tríd an slacán. Nuair a théann an
+liathróid síos go dtí leibhéal an tslacáin, is féidir linn [[seic breise|extra check]] a dhéanamh an
+bhfuil an slacán faoin liathróid, agus má tá sé casfaimid an liathróid timpeall díreach mar a
+dhéanaimid nuair a bhuaileann sé imill an stáitse.
+
+Chun seic a dhéanamh an bhfuil an slacán faoin liathróid, is féidir linn seiceáil an bhfuil lár an
+liathróid (`x_tuartha`) idir `x_slacáin` agus `x_slacáin + lthd_slacáin`.
```

Comment:
Beidh fadhb anseo má tá dx/dy > lthd_slacáin!

Reply:
Tá an ceart agat, ach bheadh sé i bhfad níos casta an slonn ceart a chruthú agus a mhíniú. Tá an seic seo cgl i gcomhthéacs an teagaisc i mo thuairim